Postgres fs impl -- WIP

Gastove opened the pull request:
Opening this PR for general eyes-on-the-project purposes -- this once again switches a decent amount of stuff around. Mostly so far: swaps bidi for compojure.

Gastove committed:
Exclude conflicting (but unused) clojure function

Gastove committed:
Re-assemble images on read

Gastove committed:
Update schema to support bytea, mime-type

Gastove committed:
Primary re-wiring/finishing of the postgres backend

Gastove committed:
Wire up pg backend, fix tests

Gastove committed:
Fix tests for real

Gastove committed:
Merge branch 'master' into postgres-fs-impl

Gastove committed:
Bump versions; swap compojure for bidi

Gastove committed:
Add a docs config prefix

Gastove committed:
Closes #63

Gastove committed:
Switch to Compojure for routes

Gastove committed:
Switch FS backends to use new compojure routes

Gastove committed:
NS Fiddlings

Gastove committed:
Derp (update NS)

jarofghosts commented on src/doctopus/db/schema.clj:
```diff
@@ -87,8 +89,20 @@
   (let [update-sql "UPDATE documents SET search_vector = to_tsvector('english', name || ' ' || body)"
-        idx-sql "CREATE INDEX fts_idx ON documents USING GIN(search_vector)"]
-    (log/info "Creating FTS Index for the documents table")
+        idx-sql (->> ["DO $$"
```
not to tack on more deps, but wondering if you have looked at [yesql](https://github.com/krisajenkins/yesql)

Gastove replied:
Ohhh, that is very pleasant! Hadn't met that one before. Might just put that on my to-implement list.

Gastove replied:
I think moving all the raw SQL in to yesql sounds like a lovely improvement. For now, everywhere we _are_ using Korma is veeeeeery simple and inoffensive, so I'm inclined to leave it in place. (Besides, despite yasql's author's grousing, Korma is _pretty nice_ for little things.)

Gastove committed:
Fix function name

fardog commented:
preliminary review says :+1: (please note: i merged in master to do the test against the latest js stuff, and i only tested the fs impl to ensure nothing broke, did not test postgres since it's not ready)

Gastove committed:
Recommend trampoline

Gastove committed:
This is a better name

Gastove committed:
Start using count of records to see if we have HTML

Gastove committed:
Mostly complete postgres backend

Returns things with real, mostly-correctly-set response bodies from the DB!

Gastove committed:
Mock out, fix later

Gastove committed:
Merge branch 'master' into postgres-fs-impl

Gastove committed:
Include mime type in query results

Gastove committed:
Cruft

Gastove committed:
Even closer to working

Gastove committed:
STOP READING FILES

Gastove committed:
tweak logging

Gastove committed:
JDBC CAN'T CAST KEYWORD TYPES

THIS WHOLE PILE HAS BEEN BLOWING UP BECAUSE I USED A KEYWORD SO NOW ITS
A STRING AND IT WORKS JFC ARE YOU HAPPY NOW JDBC EH EH i'm getting a whiskey

Gastove committed:
Mostly remove debug logging

Gastove committed:
tidy oop

Gastove committed:
Correct names

Gastove committed:
Merge branch 'master' into postgres-fs-impl

Gastove committed:
Closes #66

Gastove committed:
Changed my mind

No slash in the prefix; other places can add it as needed, instead of removing

Gastove committed:
squash! Changed my mind

Gastove committed:
Get this thing back in to shape

Needs to change to writing things in to
root/`docs-uri-prefix`/tentacle-name to support the URI scheme

Gastove committed:
Clean up temp-fs too

squash! Clean up temp-fs too

squash! squash! Clean up temp-fs too

Gastove committed:
For now, disable

Gastove commented:
This should be basically good to go! 

Closes #51 
Closes #63

Gastove committed:
Update config-local template

Gastove committed:
Disable FS backends, default to PG

Gastove committed:
STUFF

1. Strip out the rest of the temp-fs, especially from tests
2. Separate `bootstrap` and `bootstrap-schema`, to generate an empty DB
for testing.

Gastove committed:
Missed one

Gastove committed:
AT LAST

fardog commented:
:+1:
